backend/dataall/modules/shares_base/services/sharing_service.py

```
@@ -152,87 +173,106 @@ def revoke_share(cls, engine: Engine, share_uri: str) -> bool:
+                                    session,
+                                    share_uri,
+                                    old_status=ShareItemStatus.Revoke_Approved.value,
+                                    new_status=ShareItemStatus.Revoke_Failed.value,
+                                    share_item_type=processor.type,
+                                )
+                                ShareStatusRepository.update_share_item_status_batch(
+                                    session,
+                                    share_uri,
+                                    old_status=ShareItemStatus.Revoke_In_Progress.value,
+                                    new_status=ShareItemStatus.Revoke_Failed.value,
+                                    share_item_type=processor.type,
+                                )
+                                service_exceptions.append(str(e))
+                                revoke_successful = False
+                except Exception as e:
+                    log.error(f'Error occurred during share revoking: {e}')
+                    new_share_item_state = share_item_sm.run_transition(ShareItemActions.Failure.value)
+                    share_item_sm.update_state(session, share_data.share.shareUri, new_share_item_state)
+                    revoke_successful = False
+                    service_exceptions.append(str(e))
+                finally:
+                    existing_pending_items = ShareStatusRepository.check_pending_share_items(session, share_uri)
+                    if existing_pending_items:
+                        new_share_state = share_sm.run_transition(ShareObjectActions.FinishPending.value)
+                    else:
+                        new_share_state = share_sm.run_transition(ShareObjectActions.Finish.value)
+                    share_sm.update_state(session, share_data.share, new_share_state)
+
+        except Exception as e:
```

should you rethrow the exception as in the original code?

---

In the original code as well, the exception was not thrown. Can you please let me know if I am missing something

---

in the original code if there was an exception between lines `[156,177]` if would have bubble up. With your change it will be handled in the block [259,262]